app/src/components/OrderBookSnapshot.tsx

```diff
@@ -396,24 +412,31 @@ class OrderBookSnapshot extends Component<WithStyles, State> {
                         && (
                             <div>
                                 <Card className={classes.graphCard}>
-                                    {topOfBookItems.length === 0
-                                        ? (
-                                            <Typography
-                                                className={classes.noDataMessage}
-                                                variant={'body1'}
-                                                color={'textPrimary'}
-                                            >
-                                                {'Could not retrieve graph for this day.'}
-                                            </Typography>
-                                        )
-                                        : (
-                                            <TopOfBookGraphWrapper
-                                                className={classes.graph}
-                                                onTimeSelect={this.handleSelectGraphDateTime}
-                                                selectedDateTimeNano={selectedDateTimeNano}
-                                                topOfBookItems={topOfBookItems}
+                                    {graphUnavailable && (
+                                        <Typography
+                                            className={classes.noDataMessage}
+                                            variant={'body1'}
+                                            color={'textPrimary'}
+                                        >
+                                            {'Could not retrieve graph for this day.'}
+                                        </Typography>
+                                    )}
+                                    {loadingGraph && (
+                                        <div className={classes.graphLoader}>
+                                            <CustomLoader
+                                                size={'5rem'}
+                                                type={'circular'}
                                             />
-                                        )}
+                                        </div>
+                                    )}
+                                    { !loadingGraph && !graphUnavailable && topOfBookItems.length !== 0 && (
```

Comment:
Is there ever a case where `topOfBookItems.length` is 0? If so, should we handle it?

Reply:
Hmm, I don't think so, unless we somehow request 0 data points. I can remove that condition.

Reply:
Wait, actually, the initial state of OrderBookSnapshot has the topOfBookItems set to empty

Reply:
So I guess we simply don't want to display anything in that case?

Reply:
Yeah, we don't want to display anything. If you give an empty array of data to the chart library, it throws an error, so I think it's safest to keep this condition check